src/main/java/com/couchbase/lite/util/Utils.java

```diff
@@ -81,6 +84,14 @@ public static boolean isPermanentError(int code) {
         return (code >= 400 && code <= 405) || (code == 407) || (code >= 409 && code <= 499);
     }
 
+    public static boolean isTransientErrorWithSpecialCases(Response response) {
+        return isTransientErrorWithSpecialCases(response.code());
+    }
+
+    public static boolean isTransientErrorWithSpecialCases(int code) {
+        return code == 406 || code == 408 || isTransientError(code);
```

Comment:
http://docs.couchdb.org/en/latest/api/basics.html#http-status-codes

> 406 - Not Acceptable
> The requested content type is not supported by the server.

In general, this should be permanent error.